ui/app/routes/observability/functions/$function_name/route.tsx

```diff
@@ -116,7 +153,84 @@ export async function loader({ request, params }: Route.LoaderArgs) {
     variantPerformancesPromise,
     variantCountsPromise,
     variantThroughputPromise,
+    feedbackTimeseriesPromise,
   ]);
+
+  // Compute optimal probabilities for track_and_stop experimentation
+  let optimal_probabilities: Record<string, number> | undefined;
+  if (function_config.experimentation.type === "track_and_stop") {
+    try {
+      const dbClient = await getNativeDatabaseClient();
+      const experimentationConfig = function_config.experimentation;
+      if (experimentationConfig.type !== "track_and_stop") {
+        throw new Error("Expected track_and_stop experimentation type");
+      }
+      const metric_config = config.metrics[experimentationConfig.metric];
+
+      if (metric_config) {
+        const feedback = await dbClient.getFeedbackByVariant({
+          metric_name: experimentationConfig.metric,
+          function_name,
+          variant_names: experimentationConfig.candidate_variants,
+        });
+
+        // Build feedback count map
+        const feedbackCounts = new Map(
+          feedback.map((f) => [f.variant_name, f.count]),
+        );
+
+        // Separate nursery and bandit variants
+        const K = experimentationConfig.candidate_variants.length;
+        const nurseryVariants = experimentationConfig.candidate_variants.filter(
+          (v) =>
+            (feedbackCounts.get(v) || 0) <
+            Number(experimentationConfig.min_samples_per_variant),
+        );
+        const banditVariants = experimentationConfig.candidate_variants.filter(
+          (v) =>
+            (feedbackCounts.get(v) || 0) >=
+            Number(experimentationConfig.min_samples_per_variant),
+        );
+
+        const B = banditVariants.length;
+
+        // Initialize probabilities
+        optimal_probabilities = {};
+
+        // Assign 1/K to each nursery variant
+        for (const variant of nurseryVariants) {
+          optimal_probabilities[variant] = 1 / K;
+        }
+
+        // Compute and scale optimal probabilities for bandit variants
+        if (B > 0) {
+          const banditFeedback = feedback.filter((f) =>
+            banditVariants.includes(f.variant_name),
+          );
+
+          if (banditFeedback.length > 0) {
+            const banditOptimalProbs = estimateTrackAndStopOptimalProbabilities(
+              {
+                feedback: banditFeedback,
+                epsilon: experimentationConfig.epsilon,
+                metric_optimize: metric_config.optimize,
+              },
+            );
+
+            // Scale bandit probabilities by B/K
+            const scale = B / K;
+            for (const [variant, prob] of Object.entries(banditOptimalProbs)) {
+              optimal_probabilities[variant] = prob * scale;
+            }
+          }
+        }
```

Comment:
I don't like that this is reimplemented (very error prone). Please open an issue assigned to yourself to actually go back and expose whatever we need to compute this using exactly the same code in prod.